include/eve/module/complex/regular/traits.hpp

```diff
@@ -0,0 +1,58 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+#include <eve/concept/value.hpp>
+#include <eve/traits.hpp>
+
+namespace eve
+{
+  template<floating_scalar_value Type> struct complex;
+
+  template<typename T> struct as_complex;
+
+  template<floating_scalar_value T> struct as_complex<T>
+  {
+    using type = eve::complex<T>;
+  };
+
+  template<floating_scalar_value T> struct as_complex<complex<T>>
+  {
+    using type = eve::complex<T>;
+  };
+
+  template<floating_simd_value T> struct as_complex<T>
+  {
+    using type = eve::wide<eve::complex<element_type_t<T>>, cardinal_t<T>>;
+  };
```

Comment:
Do we need these?

Reply:
For a given generic type T that you want to turn into a complex, complex<T> can't work if T is wide<X,N>